src/models/Country.php

```diff
@@ -123,91 +126,134 @@ function($a, $b) {
   // and things looking like shit in the map. See issue #20.
   public static async function genAllEnabledCountriesForMap(
+    bool $refresh = false,
   ): Awaitable<array<Country>> {
-    return await self::genAll(
-      'SELECT * FROM countries WHERE enabled = 1 ORDER BY CHAR_LENGTH(d)',
-      self::MC_KEY_ALL_ENABLED_COUNTRIES_FOR_MAP,
+    $mc_result = self::getMCRecords('ALL_ENABLED_COUNTRIES_FOR_MAP');
+    if (!$mc_result || count($mc_result) === 0 || $refresh) {
+      $all_countries = await self::genAll(
+        'SELECT * FROM countries WHERE enabled = 1 ORDER BY CHAR_LENGTH(d)',
+      );
+      self::setMCRecords('ALL_ENABLED_COUNTRIES_FOR_MAP', $all_countries);
+    }
+    $all_countries = self::getMCRecords('ALL_ENABLED_COUNTRIES_FOR_MAP');
+    invariant(
+      is_array($all_countries),
+      'all_countries should be an array of Country',
     );
+    return $all_countries;
   }
 
   // All enabled and unused countries
   public static async function genAllAvailableCountries(
+    bool $refresh = false,
   ): Awaitable<array<Country>> {
-    $db = await self::genDb();
-
-    $result = await $db->queryf(
-      'SELECT * FROM countries WHERE enabled = 1 AND used = 0',
-    );
-
-    $countries = array();
-    foreach ($result->mapRows() as $row) {
-      $countries[] = await self::countryFromRow($row->toArray());
+    $mc_result = self::getMCRecords('ALL_AVAILABLE_COUNTRIES');
+    if (!$mc_result || count($mc_result) === 0 || $refresh) {
+      $all_countries = await self::genAll(
+        'SELECT * FROM countries WHERE enabled = 1 AND used = 0',
+      );
+      self::setMCRecords('ALL_AVAILABLE_COUNTRIES', $all_countries);
     }
-
-    usort(
-      $countries,
-      function($a, $b) {
-        return strcmp($a->name, $b->name);
-      },
+    $all_countries = self::getMCRecords('ALL_AVAILABLE_COUNTRIES');
+    invariant(
+      is_array($all_countries),
+      'all_countries should be an array of Country',
     );
-
-    return $countries;
+    return $all_countries;
   }
 
   // Check if country is in an active level
   public static async function genIsActiveLevel(
     int $country_id,
   ): Awaitable<bool> {
-    $db = await self::genDb();
-
-    $result = await $db->queryf(
-      'SELECT COUNT(*) FROM levels WHERE entity_id = %d AND active = 1',
-      $country_id,
-    );
-
-    invariant($result->numRows() === 1, 'Expected exactly one result');
-    return intval(firstx($result->mapRows())['COUNT(*)']) > 0;
+    return Level::genWhoUses($country_id) != null;
   }
 
   // Get a country by id.
-  public static async function gen(int $country_id): Awaitable<Country> {
-    $db = await self::genDb();
-
-    $result = await $db->queryf(
-      'SELECT * FROM countries WHERE id = %d LIMIT 1',
-      $country_id,
+  /* HH_IGNORE_ERROR[4110]: Claims - It is incompatible with void because this async function implicitly returns Awaitable<void>, yet this returns Awaitable<Country> and the type is checked on line 230 */
```

Comment:
The reason it's saying this, is because if the final `if` statement is false, the function will end without returning a value.